tests/samples_test.py

```diff
@@ -0,0 +1,448 @@
+    new_domain = samples.domain * 5 + 10
+    samples.domain = new_domain
+    assert numpy.array_equal(samples.domain, new_domain)
+    assert numpy.allclose(samples.points, new_points, atol=1e-10)
+
+
+@pytest.mark.parametrize(
+    "nested_samples,num_per_level,start_level,end_level",
+    [
+        (
+            smolyay.samples.NestedClenshawCurtisPointSet([-1, 1], 0),
+            [1],
+            [0],
+            [1],
+        ),
+        (
+            smolyay.samples.NestedClenshawCurtisPointSet([-1, 1], 1),
+            [1, 2],
+            [0, 1],
+            [1, 3],
+        ),
+        (
+            smolyay.samples.NestedClenshawCurtisPointSet([-1, 1], 2),
+            [1, 2, 2],
+            [0, 1, 3],
+            [1, 3, 5],
+        ),
+        (
+            smolyay.samples.NestedClenshawCurtisPointSet([-1, 1], 3),
+            [1, 2, 2, 4],
+            [0, 1, 3, 5],
+            [1, 3, 5, 9],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 0),
+            [1],
+            [0],
+            [1],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 1),
+            [1, 2],
+            [0, 1],
+            [1, 3],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 2),
+            [1, 2, 2],
+            [0, 1, 3],
+            [1, 3, 5],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 3),
+            [1, 2, 2, 4],
+            [0, 1, 3, 5],
+            [1, 3, 5, 9],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 4),
+            [1, 2, 2, 4, 0],
+            [0, 1, 3, 5, 9],
+            [1, 3, 5, 9, 9],
+        ),
+        (
+            smolyay.samples.SlowNestedClenshawCurtisPointSet([-1, 1], 5),
+            [1, 2, 2, 4, 0, 8],
+            [0, 1, 3, 5, 9, 9],
+            [1, 3, 5, 9, 9, 17],
+        ),
+        (
+            smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 0),
+            [1],
+            [0],
+            [1],
+        ),
+        (
+            smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 1),
+            [1, 2],
+            [0, 1],
+            [1, 3],
+        ),
+        (
+            smolyay.samples.NestedTrigonometricPointSet([0, 2 * numpy.pi], 2),
+            [1, 2, 6],
+            [0, 1, 3],
+            [1, 3, 9],
+        ),
+    ],
+    ids=nested_sample_ids,
+)
+def test_nested_levels(nested_samples, num_per_level, start_level, end_level):
+    """test number of points per level and start and end indexes"""
+    assert numpy.array_equal(nested_samples.num_per_level, num_per_level)
+    assert numpy.array_equal(nested_samples.start_level, start_level)
+    assert numpy.array_equal(nested_samples.end_level, end_level)
+    check_level = nested_samples.max_level
+    nested_samples.max_level = 6
+    assert check_level != 6
```

Comment:
I'm not really sure what this is doing / how much it helps. If you want to check that changing `max_level` is working right, I would probably do a separate test (parametrized on just 1 instance of each type of nested set) and make sure that the number of points and the size of all the level-based variables changes correctly when you change the level.

Reply:
That is because it was a partial implementation of a change that I thought I hadn't included in the commit because I was planning to be a full implementation in the morning.